pallets/capacity/src/types.rs

```diff
@@ -35,12 +26,50 @@ pub struct UnlockChunk<Balance, BlockNumber> {
 	/// Block number at which point funds are unlocked.
 	thaw_at: BlockNumber,
 }
+
+impl<T: Config> StakingAccountDetails<T> {
+	/// Increases total and active balances by an amount.
+	pub fn increase_by(&mut self, amount: BalanceOf<T>) {
+		self.total = amount.saturating_add(self.total);
+		self.active = amount.saturating_add(self.active);
+	}
+
+	/// Calculates a stakable amount from a propose amount.
+	pub fn get_stakable_amount_for(
+		&self,
+		staker: &T::AccountId,
+		proposed_amount: BalanceOf<T>,
+	) -> BalanceOf<T> {
+		let account_balance = T::Currency::free_balance(&staker);
+		let available_staking_balance = account_balance.saturating_sub(self.total);
+		available_staking_balance.min(proposed_amount)
+	}
+}
+
 impl<T: Config> Default for StakingAccountDetails<T> {
 	fn default() -> Self {
 		Self { active: Zero::zero(), total: Zero::zero(), unlocking: Default::default() }
 	}
 }
 
+/// Details about the total token amount targeted to an MSA.
+/// The Capacity that the target will receive.
+#[derive(PartialEq, Eq, Default, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct StakingTargetDetails<Balance> {
+	/// The total amount of tokens that have been targeted to the MSA.
+	pub amount: Balance,
+	/// The total Capacity that an MSA received.
+	pub capacity: Balance,
+}
+
+impl<Balance: Saturating + Copy> StakingTargetDetails<Balance> {
+	/// Increase an MSA target Staking total and Capacity amount.
+	pub fn increase_by(&mut self, amount: Balance, capacity: Balance) {
+		self.amount = amount.saturating_add(self.amount);
+		self.capacity = capacity.saturating_add(self.capacity);
```

Comment:
Overflow?

Reply:
Should we call [ensure_add](https://paritytech.github.io/substrate/master/sp_arithmetic/traits/trait.EnsureAdd.html) instead?  I don't see `ensure_add` used in substrate though; it's always `saturating_add`. Similarly with *_sub.  What are the implications of this silently adding to the point of saturation?

Reply:
After looking at it seemed easier to implement checked_add.